manifests/init.pp

```diff
@@ -10,6 +10,7 @@
   String            $ensure_service                   = $squid::params::ensure_service,
   Pattern[/\d+ KB/] $maximum_object_size_in_memory    = $squid::params::maximum_object_size_in_memory,
   String            $package_name                     = $squid::params::package_name,
+  String            $package_ensure                   = $squid::params::package_ensure,
```

Comment:
more accurate `Variant[Enum['present','latest'], Pattern[/^\d.*/]]`

Reply:
I've created a custom type to cover all enum cases (present, latest, absent and purged) and also the Pattern case.